astroid/inference.py

```diff
@@ -441,6 +442,7 @@ def infer_subscript(
             except (
                 AstroidTypeError,
                 AstroidIndexError,
+                AstroidValueError,
```

Comment:
Thoughts on simplifying this to `AstroidError`?

Reply:
AstroidValueError make sense imo as we reraise a ValueError.

Reply:
Sorry, I meant simplifying all three lines (since they all derive from `AstroidError`)

Reply:
I'm happy to leave as-is

Reply:
Hmm sorry I missed that. Yeah I think it's pretty safe to simplify the ``except`` (It's probably going to avoid problem in the future too).

Reply:
No need to apologise, when I look at it now it's not clear that the fact that I highlighted just the three lines is significant. It was clear to me when I highlighted them because that was an intentional action, but it's not clear when viewing the PR since it just looks like context. When I said "this", I should have said "these three lines".

Reply:
I think I'll leave it, `AstroidError` feels too broad. Maybe that means there's a "missing" exception type that could be the base of `Astroid{Index,Value,Type}Error` etc.? Maybe they should derive from `ResolveError` instead?

Reply:
That make sense, we don't want to hide genuine problem in astroid by catching an exception that is too general. Maybe we leave it like this for now and we do a refactor in another MR ?